### ISSUE TITLE
ICE with statics containing #[simd] types with length not a power-of-two

### ISSUE THREAD

**Issue by huonw** [rust-lang/rust#17170]
```rust
#![feature(simd)]

#[simd]
struct T(f64, f64, f64);

static X: T = T(0.0, 0.0, 0.0);

fn main() {
    let _ = X;
}
```

```
{ double, double, double } zeroinitializer
<3 x double> undef
error: internal compiler error: const expr(19: T(0.0, 0.0, 0.0)) of type T has size 24 instead of 32
note: the compiler hit an unexpected failure path. this is a bug.
note: we would appreciate a bug report: http://doc.rust-lang.org/complement-bugreport.html
note: run with `RUST_BACKTRACE=1` for a backtrace
task 'rustc' failed at 'Box<Any>', /home/rustbuild/src/rust-buildbot/slave/nightly-linux/build/src/libsyntax/ast_util.rs:776
```

Similarly 5 fields fails with `has size 40 instead of 64`.

**Comment by simnalamburt** [comment]
I think the right-hand side of the statement below should be interpreted as `<3 x double>` not `{ double, double, double }`. Let's figure this out!

```rust
static X: T = T(0.0, 0.0, 0.0);
```

**Comment by simnalamburt** [comment]
This issue has been fixed, tested with `rustc 1.0.0-nightly (07560d233 2015-04-20)`. Please close this.

**Comment by huonw** [comment]
Tagged as needstest, thanks @simnalamburt!
